packages/legend-query-builder/src/stores/QueryBuilderResultState.ts

```diff
@@ -260,20 +261,31 @@ export class QueryBuilderResultState {
         );
       }
     } catch (error) {
-      assertErrorThrown(error);
-      this.queryBuilderState.applicationStore.logService.error(
-        LogEvent.create(GRAPH_MANAGER_EVENT.EXECUTION_FAILURE),
-        error,
-      );
-      this.queryBuilderState.applicationStore.notificationService.notifyError(
-        error,
-      );
+      // When the query is canceled on the server's side by calling the API, there will be an execution failure error.
+      // We don't notify users about this success or failure.
+      // However, we should notify users about any failure from the current query that is being executed.
+      if (this.queryRunPromise === promise) {
```

Comment:
// When user cancels the query by calling the `cancelQuery` api, it will throw an exeuction failure error. 
// For now, we don't want to notify users about this failure. Therefore we check to ensure the promise is still the same one.
// When cancelled the query, we set the `queryRunPromise` as undefined.